packages/nimble-components/src/drawer/index.ts

```diff
@@ -25,109 +24,83 @@ declare global {
  * which animates to be visible with a slide-in / slide-out animation.
  */
 // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
-export class Drawer<CloseReason = void> extends FoundationElement {
+export class Drawer<CloseReason = void> extends DialogBase<CloseReason> {
     // We want the member to match the name of the constant
     // eslint-disable-next-line @typescript-eslint/naming-convention
     public static readonly UserDismissed = UserDismissed;
 
     @attr
     public location: DrawerLocation = DrawerLocation.right;
 
-    @attr({ attribute: 'prevent-dismiss', mode: 'boolean' })
-    public preventDismiss = false;
-
-    public dialog!: ExtendedDialog;
     private closing = false;
 
-    private resolveShow?: (reason: CloseReason | UserDismissed) => void;
     private closeReason!: CloseReason | UserDismissed;
 
     /**
-     * True if the drawer is open, opening, or closing. Otherwise, false.
-     */
-    public get open(): boolean {
-        return this.resolveShow !== undefined;
-    }
-
-    /**
-     * Opens the drawer
-     * @returns Promise that is resolved when the drawer finishes closing. The value of the resolved
-     * Promise is the reason value passed to the close() method, or UserDismissed if the drawer was
-     * closed via the ESC key.
-     */
-    public async show(): Promise<CloseReason | UserDismissed> {
-        if (this.open) {
-            throw new Error('Drawer is already open');
-        }
-        this.openDialog();
-        return new Promise((resolve, _reject) => {
-            this.resolveShow = resolve;
-        });
-    }
-
-    /**
-     * Closes the drawer
-     * @param reason An optional value indicating how/why the drawer was closed.
+     * Closes the Drawer
+     * @param reason An optional value indicating how/why the Drawer was closed.
      */
-    public close(reason: CloseReason): void {
+    public override close(reason: CloseReason): void {
```

Comment:
I am just starting to look at it. My first impression is that if around half the methods of the base class are overridden without calling super for the default behavior then I'm not sure how much value we are getting in the abstraction / maybe the abstraction is misaligned. I'll have to review it more closely to understand the goals of the split / responsibilities of the base class.

Reply:
@m-akinc to prevent blocking the functional aspect of the change, can we split the refactor and event handling updates into separate PRs?

Reply:
Moved the functional aspects into PR #1848

Reply:
I agree after scanning that the benefit is less compelling if only one of the derived classes uses many of the methods in the base class. Some general strategies around this:
1. don't put those methods in the base class. Each derived class has its own implementation. If, after doing this, the base class is mostly empty or isn't cohesive (e.g. has an `open` but not a `close`) then we could just get rid of it.
2. narrow the scope of those methods to only include the common functionality, then have the derived classes call super and then add their custom functionality
3. make the base class handle both cases and choose between them based on some configuration of the object (don't love this option, probably a mixin or larger class hierarchy would make more sense, but none of that feels right for this tiny class hierarchy)
4. find a way to make the behaviors identical so they can be handled entirely by the base class

@m-akinc wanna take a pass at applying one or more of these strategies?

Reply:
One thought I had is having Modal as the base class, the promise workflow is all handled there (we don't need implementation specific open and close value tracking and promise resolution). So things like resolveShow are private there and the dialog reference is protected (calling showModal on a dialog element is managed in the base class).

And then the base class has lifecycle methods that can be overridden / called and some state tracking. 

Something like states: {opening, open, closing, closed}
and methods (some protected) for the transitions: open, finishOpening, close, finishClosing.

So a dialog without animations in close will immediately call finishClosing while one with other work like animations can call finishClosing later. Think the logic will be clearer too tracking a state vs implied knowledge based on whether resolveShow is set or not.

Reply:
> the dialog reference is protected

I guess it needs to be public for `ref` to work in the template.

Reply:
Looking a lot better. The thing I want to emphasize was the goal "So things like resolveShow are private there" and "And then the base class has lifecycle methods that can be overridden / called and some state tracking."

The intention is that the ModalState tracking is completely handled in the base class and private. All modals go through the state transitions (off the top, didn't double check what was implemented) closed -> opening -> open -> closing -> closed. And the transitions between states are handled by methods that do the transitions finishOpening, finishClosing, etc.

Previously `resolveShow` represented the state of the control and now `state` represents the state of control. Either way the goal of my statement is that the state management is private / encapsulated in Modal.

Reply:
> The intention is that the ModalState tracking is completely handled in the base class and private.

I had tried to do that, but it didn't seem like it would work. On second look, we can make it work with a couple assumptions:
1. Derived classes can override the `startOpening/Closing` methods, but should not call them (except from inside an override). I'm having `Modal` set the state immediately before calling those methods, so the state won't be set correctly if the derived class calls `startOpening` somewhere else. 
2. The base implementation for `finishOpening/Closing` sets the state, so if a derived class overrides one of those functions, it _must_ call the base implementation (or the state won't get updated properly).

I wish there were a nice way to enforce these assumptions, but instead I've just added comments to the base implementations.